pkg/health/server/server.go

```diff
@@ -264,48 +241,27 @@ func (s *Server) Shutdown() {
 	for i := range s.tcpServers {
 		s.tcpServers[i].Shutdown()
 	}
-	if !s.Config.Passive {
-		s.Server.Shutdown()
-	}
+	s.Server.Shutdown()
 }
 
-func enableAPI(opt AdminOption, tcpPort int) bool {
-	switch opt {
-	case AdminOptionAny:
-		return true
-	case AdminOptionUnix:
-		return tcpPort == 0
-	default:
-		return false
-	}
-}
-
-// newServer instantiates a new instance of the API that serves the health
-// API on the specified port. If tcpPort is 0, then a unix socket is opened
-// which serves the entire API. If a tcpPort is specified, then it returns
-// a server which only answers get requests for the root URL "/".
-func (s *Server) newServer(spec *loads.Document, tcpPort int) *healthApi.Server {
+// newServer instantiates a new instance of the health API server on the
+// defaults unix socket.
+func (s *Server) newServer(spec *loads.Document) *healthApi.Server {
 	api := restapi.NewCiliumHealthAPI(spec)
 	api.Logger = log.Printf
 
-	// /hello
+	// /hello - this is still used to check if admin API is up
```

Comment:
Is it? 

`/hello` should be used for network-level pings. There's no need to do a ping over the unix socket, is there?

EDIT: I see you note that in the PR message, fair enough, that can be dropped in a subsequent PR.